### PR TITLE
Joca96/565 implement headless combobox

### PR DESCRIPTION
**Pull request by JoCa96**
Relates to #565 

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)

### CODE REVIEW

File: packages/headless/src/utils/keyboard.ts [hunk 1]
```diff
@@ -1,3 +1,26 @@
...
+  | Partial<Pick<KeyboardEvent, "altKey" | "key" | "ctrlKey" | "metaKey" | "shiftKey" | "code">>;
+
+/**
+ * Check if a specified key was pressed.
+ * @param event The KeyboardEvent
+ * @param key The key, either the [key property](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) as a string (e.g. "m")
+ * or an object with the relevant key parameters, e.g. `{ key: "m", altKey: true }`
+ * @returns true, if the key was pressed with the specified parameters
+ */
+export const wasKeyPressed = (event: KeyboardEvent, key: WasKeyPressedOption) => {
```
Comment (larsrickert) [inline packages/headless/src/utils/keyboard.ts:14]: Why is this needed? Couldn't we just check the pressed key?

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/headless/package.json [hunk 2]
```diff
@@ -24,11 +24,14 @@
   },
   "scripts": {
     "build": "vue-tsc --build --force",
-    "test": "vitest",
+    "test": "vitest --environment jsdom",
```
[the remark this answers is not in the document]
Comment (JoCa96) [reply packages/headless/package.json:27]: sounds good

File: packages/headless/src/utils/keyboard.ts (hunk 1 above)
Comment (JoCa96) [reply packages/headless/src/utils/keyboard.ts:14]: I want to have a convenient way to check for a variety of possible key combinations without a dozen `if (...) else` blocks, e.g. see https://github.com/SchwarzIT/onyx/pull/1160/files#diff-e20cf7a25727ff2875e3ea6f35553d0cef6f794a375ed5b648ce98ba4add672aR24

**Review by larsrickert — APPROVED** [review]
